src/core/generator/json/http_directive_generator.rs

```diff
@@ -50,36 +50,96 @@ impl<'a> UrlUtility<'a> {
 
 pub struct HttpDirectiveGenerator<'a> {
     url: &'a Url,
+    route: Option<&'a String>,
```

Comment:
```suggestion
    route: Option<&'a Rest>,
```

We have already implemented a lot of machinery for `Rest` we should use that. 
Side Note: May be rename `Rest` to `Route` in a separate PR.

Reply:
The implementation here is a bit different than Rest, from my understanding Rest builds a type map of user defined types. 

[](https://tailcall.run/docs/tailcall-dsl-graphql-custom-directives/#rest-directive)

where $id is already known to be an _Int!_.

In here we don't really know the exact type of the segment unless we try to infer it, and when we try to infer we're already at a point where we can just build the _Field_ if that make sense. 

We can however let the user explicitly define the type of $id and the configuration will change a little bit to be like : 

```json
[
      {
        "curl":{
            "src":"https://jsonplaceholder.typicode.com/posts/1",
            "fieldName":"posts",
            "rest": {
              "route": "/posts/$id",
               "variables": {
                 "$id": "Int!"
               }
            }
        }
      }
  ],
```

originally it was like this(which is a lot more simple) : 

```
[
      {
        "curl":{
            "src":"https://jsonplaceholder.typicode.com/posts/1",
            "fieldName":"posts",
            "path": "/posts/$id" // optional new configuration to describe the path
        }
      }
  ],
```